### PR TITLE
go.{mod,sum}: bump mkctr

### PR DESCRIPTION
**Pull request by tomhjp**
`go get github.com/tailscale/mkctr@main`

Pulls in changes to support a local target that only pushes a single-platform image to the machine's local image store.

Fixes tailscale/mkctr#18

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -125,10 +125,18 @@ require (
...
+	github.com/felixge/httpsnoop v1.0.4 // indirect
+	github.com/go-logr/stdr v1.2.2 // indirect
 	github.com/gobuffalo/flect v1.0.2 // indirect
 	github.com/google/gnostic-models v0.6.9-0.20230804172637-c7be7c783f49 // indirect
 	github.com/google/pprof v0.0.0-20240409012703-83162a5b38cd // indirect
 	github.com/gorilla/securecookie v1.1.2 // indirect
+	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.47.0 // indirect
+	go.opentelemetry.io/otel v1.22.0 // indirect
+	go.opentelemetry.io/otel/metric v1.22.0 // indirect
+	go.opentelemetry.io/otel/trace v1.22.0 // indirect
```
Comment (irbekrm) [inline go.mod:139]: I was a bit confused where all these new deps came from, but looks like they are all deps of the daemon package added in https://github.com/tailscale/mkctr/commit/b5d5d7e0579866bcccc48b6b318d5c8ad9ba15c6 and we don't have a way to not call this.
(I've had past issues with the otel package version compatibility as they all depend on each other, but I guess if we run into those in the future, we could start publishing mkctr binary and pull it in that way https://github.com/tailscale/tailscale/tree/main/tool)

**Review by irbekrm — APPROVED** [review]
Thank you for doing this!

I've left a dependency comment, but that's nothing actionable for this PR

File: go.mod (hunk 1 above)
Comment (tomhjp) [reply go.mod:139]: We could also remove the daemon package very easily. It's only used for writing images locally, and we already have a fallback in place by searching for the `docker` CLI.

Comment (irbekrm) [reply go.mod:139]: That sounds good - I guess that everyone will have either of those 3 tools locally 👍🏼 
(I guess we'll also use this target for e2e test scripts, unless we do something like https://github.com/ko-build/ko/pull/180)

**Comment by tomhjp** [comment]
Thanks! I'll merge this as-is for now, but very happy to proactively address the daemon dependency in a follow-up if you think we should do that sooner rather than later

**Pull request merged**